packages/playground/src/playground-db.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+defined('ABSPATH') || exit;
+
+function playground_escape_array($array)
+{
+	global $wpdb;
+	$escaped = array();
+	foreach ($array as $value) {
+		if (is_numeric($value))
+			$escaped[] = $wpdb->prepare('%d', $value);
+		else
+			$escaped[] = $wpdb->prepare('%s', $value);
+	}
+	return implode(',', $escaped);
+}
+
+function playground_dump_db($zip)
+{
+	global $wpdb;
+
+	$tables   = playground_get_db_tables();
+	$sql_dump = array();
+
+	foreach ($tables as $table) {
+		array_push(
+			$sql_dump,
+			sprintf("DROP TABLE IF EXISTS `%s`;", esc_sql($table)),
+			playground_dump_db_schema($table)
+		);
+	}
+
+	foreach ($tables as $table) {
+		$records = $wpdb->get_results(
+			sprintf(
+				'SELECT * FROM `%s`',
+				esc_sql($table)
+			),
+			ARRAY_A
+		);
+
+		if ($wpdb->last_error) {
+			error_log($wpdb->last_error);
+			continue;
+		}
+
+		foreach ($records as $record) {
+			array_push(
+				$sql_dump,
+				sprintf(
+					'INSERT INTO `%1$s` (%2$s) VALUES (%3$s);',
```

Comment:
Ditto the question from above – could this sprintf be replaced with `$wpdb->prepare()`?

Reply:
No, just like with table names `$wpdb->prepare()`  would wrap params in quotes.

Reply:
Yes, they break queries. 

Examples
- `SELECT * FROM 'wp_posts'`
- `INSERT INTO 'wp_options' ('1,"test","yes"')`

Reply:
Ah, they're singular quotes, not double quotes. Gotcha!